src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -212,8 +213,22 @@ object Scanners {
     /** A buffer for comments */
     val commentBuf = new StringBuilder
 
+    private def handleMigration(keyword: Token): Token =
+      if (!isScala2Mode) keyword
+      else if (keyword == INLINE) treatAsIdent()
```

Comment:
This means that the following doesn't work with `-language:Scala2` because `inline` is treated as an identifier:
```scala
inline def x = ...
```
you have to use `@inline` instead. We can fix this by only treating `INLINE` specially if the previous token is `AT`.